pandas/core/groupby/generic.py

```diff
@@ -1283,7 +1282,7 @@ def _wrap_applied_output_series(
         # as we are stacking can easily have object dtypes here
         so = self._selected_obj
         if so.ndim == 2 and so.dtypes.apply(needs_i8_conversion).any():
-            result = _recast_datetimelike_result(result)
+            result = result._convert(datetime=True)
```

Comment:
```.apply(to_datetime)``` ?

Reply:
we only want to do it on object-dtype columns which is what _convert does

Reply:
kk cool, then this is fine. ping on green.

Reply:
ping